### PR TITLE
Add damage sources

### PR DESCRIPTION
**Pull request by TheKodeToad**
Adds damage sources to the DamageSource enum.
No tests were added, because the names in the API are slightly different to the names in Minecraft (e.g. ANVIL -> FALLING_ANVIL).

Sorry if I'm creating too many pull requests.

### CODE REVIEW

File: server/src/main/java/org/loomdev/loom/util/transformer/DamageSourceTransformer.java
Comment (thatgoofydev) [inline server/src/main/java/org/loomdev/loom/util/transformer/DamageSourceTransformer.java]: Since hitting the default here means that there is a value in the API enum that is missing in the Minecraft enum. It would be better to fail explicitly here instead of silently so we can find this error as soon as possible. You could throw an exception here or log an error and return GENERIC if the exception causes the server to crash.

**Review by thatgoofydev — CHANGES_REQUESTED** [review]
Looks good.

I am just wondering if we can find any better names for HOT_FLOOR and IN_WALL. INSIDE_BLOCK might be a better name for IN_WALL? 🤷 @TheKodeToad and @nkomarn what do you guys think about changing these to more logical names and do you have any suggestions?

**Comment by TheKodeToad** [comment]
I agree. INSIDE_BLOCK is probably better than IN_WALL, because it can happen when underground.
HOT_FLOOR might be used for other hot blocks in the future though.

File: server/src/main/java/org/loomdev/loom/util/transformer/DamageSourceTransformer.java
Comment (TheKodeToad) [reply server/src/main/java/org/loomdev/loom/util/transformer/DamageSourceTransformer.java]: Plugins should never be able to crash the server.

**Comment by TheKodeToad** [comment]
I can't find the requested change.

**Comment by thatgoofydev** [comment]
> I can't find the requested change.

image
This was the requested change (exception instead of GENERIC), but you already changed it 😄

**Comment by TheKodeToad** [comment]
How do I fix it if it's already fixed?

**Comment by thatgoofydev** [comment]
> How do I fix it if it's already fixed?

you don't.. you wait for another review. But I am waiting if @nkomarn want to add something to the name changes or not.

File: api/src/main/java/org/loomdev/api/entity/damage/DamageSource.java
Comment (nkomarn) [inline api/src/main/java/org/loomdev/api/entity/damage/DamageSource.java]: This should probably be "DRY_OUT"

**Comment by nkomarn** [comment]
> Looks good.
> 
> I am just wondering if we can find any better names for HOT_FLOOR and IN_WALL. INSIDE_BLOCK might be a better name for IN_WALL? shrug @TheKodeToad and @nkomarn what do you guys think about changing these to more logical names and do you have any suggestions?

I think "IN_WALL" is fine, considering vanilla death messages reference "suffocated in a wall." "HOT_FLOOR" can be confusing, but I'm not really sure if there's anything much better. I'm okay with all of the names we have at the moment, aside from the minor comment I made on "DRYOUT."

**Comment by TheKodeToad** [comment]
I already changed IN_WALL to INSIDE_BLOCK.

**Comment by TheKodeToad** [comment]
I don't know what I'm meant to do now. The only requested change has already been fixed.

**Review by thatgoofydev — APPROVED** [review]

**Pull request merged**
